src/cargo/sources/git/utils.rs

```diff
@@ -565,7 +565,14 @@ pub fn fetch(repo: &git2::Repository, url: &str,
         let mut opts = git2::FetchOptions::new();
         opts.remote_callbacks(cb)
             .download_tags(git2::AutotagOption::All);
-        try!(remote.fetch(&[refspec], Some(&mut opts), None));
-        Ok(())
+
+        match network::with_retry(cargo_config, ||{
+            remote.fetch(&[refspec], Some(&mut opts), None)
+        }){
+            Ok(_) => Ok(()),
+            Err(e) =>{
+                Err(git2::Error::from_str(&format!("{}", e)))
```

Comment:
It looks like the test pass for me locally and I can convert back to just calling try! again. I pushed it. I can revert the commit you remember why the CargoResult was turned into a Result.